pkg/core/capability/caldera/caldera.go

```diff
@@ -58,6 +59,84 @@ func (capability *calderaCapability) GetType() string {
 	return capabilityName
 }
 
+// Cleanup will attempt to remove any artifacts created by SOARCA on the Caldera installation.
+// This function is idempotent. This function will only remove artifacts tied to SOARCA-related
+// Operation objects for Operations that are finished, to prevent race conditions. Cleanup may
+// fail silently if it encounters errors.
+func cleanup(cc ICalderaConnection) {
+	staleOperations, err := cc.GetStaleOperations()
+	if err != nil {
+		log.Error("Caldera cleanup failed: unable to fetch stale operations")
+		return
+	}
+
+	if len(staleOperations) == 0 {
+		return
+	}
+
+	factSources, sourceErr := cc.GetFactSources()
+	if sourceErr != nil {
+		log.Error("Caldera cleanup failed: could not retrieve fact sources")
+	}
+
+	abilities, abilityErr := cc.GetAbilities()
+	if abilityErr != nil {
+		log.Error("Caldera cleanup failed: could not retrieve abilities")
+	}
+
+	adversaries, adversaryErr := cc.GetAdversaries()
+	if adversaryErr != nil {
+		log.Error("Caldera cleanup failed: could not retrieve adversaries")
+	}
+
+	for i := 0; i < len(staleOperations); i++ {
+		currentOperation := staleOperations[i]
+		successfulCleanup := true
+
+		for j := 0; j < len(factSources); j++ {
+			if factSources[j].Name == currentOperation.Name {
+				err := cc.DeleteFactSource(factSources[j].ID)
+				if err != nil {
+					successfulCleanup = false
+				}
+			}
+		}
+
+		for j := 0; j < len(abilities); j++ {
```

Comment:
This *shouldn't* cause issues, but theoretically an adversary references abilities, therefore the adversary should be deleted first.